### PR TITLE
IH now have a hammer

### PR DESCRIPTION
**Pull request by DarkLevelSS13**
## About The Pull Request
Adding new item to IH armory, an IH "Tactical Breaching Hammer".
Also, rebalancing wall smashing mechanic. Now the chance of successfully smashing the wall will depend on Robust stat, and you don't instantly dismantle the wall.
image
image

## Changelog
:cl: DarkLevel - code; CeUvi - sprites.
add: IH "Tactical Breaching Hammer"
balance: Rebalancing wall smashing mechanic. Make it more depends on stats of the user and item.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/tools/hammer.dm [hunk 1]
```diff
@@ -152,3 +152,19 @@
 		playsound(src, 'sound/machines/hiss.ogg', 50, 0, 0)
 		user.throw_at(target, get_dist(target, user), 1, user)
 		T.stop()
+
+/obj/item/weapon/tool/hammer/IH
+	name = "IH \"Tactical Breaching Hammer\""
```
Comment (CeUvi) [inline code/game/objects/items/weapons/tools/hammer.dm:157]: `name = "FS \"Ironhammer\" Breaching Hammer"`

**Review by CeUvi — CHANGES_REQUESTED** [review]
should change the name to what have been suggested before

File: code/game/objects/items/weapons/tools/hammer.dm (hunk 1 above)
Comment (DarkLevelSS13) [reply code/game/objects/items/weapons/tools/hammer.dm]: Done.

File: code/game/objects/items/weapons/tools/hammer.dm [hunk 2]
```diff
@@ -152,3 +152,19 @@
 		playsound(src, 'sound/machines/hiss.ogg', 50, 0, 0)
 		user.throw_at(target, get_dist(target, user), 1, user)
 		T.stop()
+
+/obj/item/weapon/tool/hammer/IH
+	name = "FS \"Ironhammer\" Breaching Hammer"
+	desc = "A modified sledgehammer produced by Frozen Star for Ironhammer forces. This tool can take down standard walls and if you robust enough, even a reinforced wall."
```
Comment (Firefox13) [inline code/game/objects/items/weapons/tools/hammer.dm:158]:
```suggestion
	desc = "A modified sledgehammer produced by Frozen Star for Ironhammer forces. This tool can take down standard walls and if the user is strong enough, reinforced walls."
```

**Comment by Firefox13** [comment]
Did you test it? What about the changes you made to how walls take damage? Where(give a picture) the hammers spawn?

**Comment by DarkLevelSS13** [comment]
> Did you test it? What about the changes you made to how walls take damage? Where(give a picture) the hammers spawn?

1) Works fine.
2) First picture.
2

**Comment by DarkLevelSS13** [comment]
The new system of penetrating walls, now works like this:
Each wall has its own gradation of hardness, depending on the material, and the greater the material hardness the harder it is to destroy the wall. This chance could not be influenced, and in fact you could not even break the reinforced walls. And if you successfully hit a wall, instead of it being by damage, it was just a random chance from one hit to destroy a wall.
It's different now. The chance of whether you can successfully hit a wall or not depends on your strength. Also, you don't hit the wall with one hit now, it takes the damage of your weapon multiplied by its structure damage modifier.

**Comment by Firefox13** [comment]
>     2. First picture.

And how you're supposed to reach it there. Having to hop over a table isn't good...

**Comment by DarkLevelSS13** [comment]
> > ```
> > 2. First picture.
> > ```
> 
> And how you're supposed to reach it there. Having to hop over a table isn't good...

Indeed )

**Comment by Wouju** [comment]
Incase of emergency, climb table

**Comment by Iglpigl** [comment]
> > > ```
> > > 2. First picture.
> > > ```
> > 
> > 
> > And how you're supposed to reach it there. Having to hop over a table isn't good...
> 
> Indeed )

You could just make the lathe on the tile hammer is on and hammer on tile lathe is on but let's see what the Firefox says.

**Comment by CeUvi** [comment]
before it gets merged, make it wearable on the suit and not on the back please @DarkLevelSS13

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**